### PR TITLE
Creating TC 25 Verify Scroll Up using 'Arrow' button and Scroll Down …

### PR DESCRIPTION
**Pull request by MostafaAbotalebGizaSystems**
…functionality:

- Adding action method "scrollToBottomPage" in FooterPage
- Adding 2 locators "scrollUpArrow_button"&"slideHeading_div" in the HomePage
- Adding action method "clickOnScrollUpArrow" in the HomePage
- Adding validation method "validateOnVisibilityOfCarouselSlideText" in the HomePage
- Creating new test class "VerifyingScrollUpUsingArrowButtonAndScrollDownTests"
- Creating new Test Data json file "ScrollingTestData.json"

### CODE REVIEW

File: src/main/java/com/gizasystems/automationexercise/pages/Footer.java [hunk 1]
```diff
@@ -3,26 +3,19 @@
 import com.shaft.driver.SHAFT;
 import io.qameta.allure.Step;
 import org.openqa.selenium.By;
-
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/Footer.java:6]: No need to delete this line.

File: src/main/java/com/gizasystems/automationexercise/pages/Footer.java [hunk 2]
```diff
@@ -3,26 +3,19 @@
...
 public class Footer {
-
     //Variables
     private SHAFT.GUI.WebDriver driver;
-
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/Footer.java:11]: No need to delete this line.

File: src/main/java/com/gizasystems/automationexercise/pages/Footer.java [hunk 3]
```diff
@@ -3,26 +3,19 @@
...
     // Locators
     private final By subscriptionText = By.xpath("//div[@class='single-widget']/h2");
     private final By subscriptionEmail = By.id("susbscribe_email");
     private final By subscribeBtn = By.id("subscribe");
     private final By successMsg = By.id("success-subscribe");
-
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/Footer.java:17]: No need to delete this line.

File: src/main/java/com/gizasystems/automationexercise/pages/Footer.java [hunk 4]
```diff
@@ -3,26 +3,19 @@
...
     // Constructor
-
     public Footer(SHAFT.GUI.WebDriver driver) {
         this.driver = driver;
     }
-
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/Footer.java:23]: No need to delete this line.

File: src/main/java/com/gizasystems/automationexercise/pages/Footer.java [hunk 5]
```diff
@@ -35,6 +28,11 @@ public Footer clickOnSubscribeButton() {
         return this;
     }
 
+    @Step("Scroll to bottom page")
+    public Footer scrollToBottomPage() {
+        driver.element().scrollToElement(subscriptionText);
+        return this;
+    }
     //////////////////// Validations \\\\\\\\\\\\\\\\\\\\
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/Footer.java:36]: Please add an empty line here between the method and the Validation section.

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java [hunk 6]
```diff
@@ -8,24 +8,28 @@ public class HomePage {
     // Variables
     private SHAFT.GUI.WebDriver driver;
     private String url = System.getProperty("baseUrl");
-
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:11]: No need to delete this line.

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java [hunk 7]
```diff
@@ -8,24 +8,28 @@ public class HomePage {
...
     // Locators
     private final By featuredItems_div = By.cssSelector("div.features_items");
     private final By recommendedItems_div = By.cssSelector("div.recommended_items");
     private final By categoryTitle_div= By.xpath("//h2[@class='title text-center']");
-
+    private final By scrollUpArrow_button = By.xpath("//i[@class='fa fa-angle-up']");
+    private final By slideHeading_div= By.xpath("(//h2)[2]");
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:16]: I believe we can have a more relative locator here.

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java [hunk 8]
```diff
@@ -8,24 +8,28 @@ public class HomePage {
...
     // Constructor
     public HomePage(SHAFT.GUI.WebDriver driver) {
         this.driver = driver;
     }
-
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:21]: No need to delete this line.

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java [hunk 9]
```diff
@@ -8,24 +8,28 @@ public class HomePage {
...
         driver.browser().navigateToURL(url);
         return this;
     }
 
+    @Step("Click on scroll up arrow")
+    public HomePage clickOnScrollUpArrow() {
+        driver.element().click(scrollUpArrow_button);
+        return this;
+    }
     //////////////////// Validations \\\\\\\\\\\\\\\\\\\\
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:33]: Please add an empty line here between the method and the Validation section.

File: src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpUsingArrowButtonAndScrollDownTests.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    @TmsLink("55512524")
+    public void verifyScrollingUpUsingArrowButtonInHomePage() {
+        new Footer(driver)
+                .scrollToBottomPage()
+                .verifySubscriptionText(testUser.getTestData("subText"));
+        new HomePage(driver)
+                .clickOnScrollUpArrow()
+                .validateOnVisibilityOfCarouselSlideText(testUser.getTestData("slideHeadingText"));
+    }
+    //////////////////// Configurations \\\\\\\\\\\\\\\\\\\\
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpUsingArrowButtonAndScrollDownTests.java:31]: Please add an empty line here between the method and the Configurations section.

File: src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpUsingArrowButtonAndScrollDownTests.java [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+import org.testng.annotations.Test;
+
+@Epic("Automation Exercise")
+@Feature("Home page navigation")
+@Story("Verify Scroll Up using 'Arrow' button and Scroll Down functionality")
+public class VerifyingScrollUpUsingArrowButtonAndScrollDownTests {
+    // Variables
+    private SHAFT.GUI.WebDriver driver;
+    private SHAFT.TestData.JSON testUser;
+    // Test Cases
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpUsingArrowButtonAndScrollDownTests.java:19]: Please add empty line here between the variables and testcases sections.

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]
Please check the conflicts.

File: src/main/java/com/gizasystems/automationexercise/pages/Footer.java (hunk 1 above)
Comment (MostafaAbotalebGizaSystems) [reply src/main/java/com/gizasystems/automationexercise/pages/Footer.java:6]: Done

File: src/main/java/com/gizasystems/automationexercise/pages/Footer.java (hunk 2 above)
Comment (MostafaAbotalebGizaSystems) [reply src/main/java/com/gizasystems/automationexercise/pages/Footer.java:11]: Done

File: src/main/java/com/gizasystems/automationexercise/pages/Footer.java (hunk 3 above)
Comment (MostafaAbotalebGizaSystems) [reply src/main/java/com/gizasystems/automationexercise/pages/Footer.java:17]: Done

File: src/main/java/com/gizasystems/automationexercise/pages/Footer.java (hunk 4 above)
Comment (MostafaAbotalebGizaSystems) [reply src/main/java/com/gizasystems/automationexercise/pages/Footer.java:23]: Done

File: src/main/java/com/gizasystems/automationexercise/pages/Footer.java (hunk 5 above)
Comment (MostafaAbotalebGizaSystems) [reply src/main/java/com/gizasystems/automationexercise/pages/Footer.java]: Done

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java (hunk 7 above)
Comment (MostafaAbotalebGizaSystems) [reply src/main/java/com/gizasystems/automationexercise/pages/HomePage.java]: Done

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java (hunk 8 above)
Comment (MostafaAbotalebGizaSystems) [reply src/main/java/com/gizasystems/automationexercise/pages/HomePage.java]: Done

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java (hunk 9 above)
Comment (MostafaAbotalebGizaSystems) [reply src/main/java/com/gizasystems/automationexercise/pages/HomePage.java]: Done

File: src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpUsingArrowButtonAndScrollDownTests.java (hunk 10 above)
Comment (MostafaAbotalebGizaSystems) [reply src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpUsingArrowButtonAndScrollDownTests.java]: Done

File: src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpUsingArrowButtonAndScrollDownTests.java (hunk 11 above)
Comment (MostafaAbotalebGizaSystems) [reply src/test/java/com/gizasystems/automationexercise/tests/VerifyingScrollUpUsingArrowButtonAndScrollDownTests.java]: Done

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java (hunk 6 above)
Comment (MostafaAbotalebGizaSystems) [reply src/main/java/com/gizasystems/automationexercise/pages/HomePage.java]: Done

**Comment by MostafaAbotalebGizaSystems** [comment]
I requested merging by mistake.
Please review the comments

**Review by MahmoudElSharkawyGS — APPROVED** [review]

**Pull request merged**